src/Expander.hs

```diff
@@ -543,7 +547,7 @@ initializeKernel = do
 
       let info =
             DatatypeInfo
-            { _datatypeArity = arity
+            { _datatypeArgKinds = replicate (fromIntegral arity) KStar
```

Comment:
this `arity` is an argument of `addDatatypePrimitive`; I guess another next step will be to push the `[Kind]` representation a bit further and use at the `addDatatypePrimitive` call sites as well.

Reply:
done.